src/Request/RequestInterface.php

```diff
@@ -9,18 +9,14 @@
     const METHOD_PATCH = 'PATCH';
     const METHOD_DELETE = 'DELETE';
 
-    /**
-     * @return string
-     */
+    const TYPE_FORM = 'application/x-www-form-urlencoded';
+    const TYPE_JSON = 'application/json';
+
     public function getMethod(): string;
 
-    /**
-     * @return array
-     */
-    public function getParams(): array;
+    public function getParams(): string;
 
-    /**
-     * @return string
-     */
     public function getEndpoint(): string;
+
+    public function getMediaType(): string;
```

Comment:
any info what the implementation should return exactly?

btw we could use also the enums here